pinot-core/src/main/java/org/apache/pinot/core/query/request/context/Expression.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.request.context;
+
+import java.util.Objects;
+import java.util.Set;
+
+
+/**
+ * The {@code Expression} class represents an expression in the query.
```

Comment:
We already have this class right? The Calcite SQL compilation path uses Expression already. Why add it again?

Reply:
Added this new Expression class for 2 reasons:
- The function is represented with the new added `FunctionInfo` class (Pinot domain class which identifies whether the function is `Aggregation` or `Transform` before sending the query to each segment)
- Calcite Expression is a Thrift object which we use for wiring. It cannot be evolved along with the query execution engine. We should have a separate POJO class to encapsulate the information needed by the query engine, and perform the pre-computation before getting into each segment if possible.

Reply:
I agree with the change, however, can we rename one of them (I am guessing we can't rename thrift one). It becomes very confusion with two classes of same name.